src/server/pfs/server/driver.go

```diff
@@ -1173,6 +1175,37 @@ func (d *driver) makeCommit(
 		}
 	}
 
+	// Set newCommitInfo.Started and possibly newCommitInfo.Finished. Enforce:
+	// 1) 'started' and 'newCommitInfo.Started' must be set
+	// 2) 'started' <= 'finished'
+	switch {
+	case started.IsZero() && finished.IsZero():
+		// set 'started' to Now() && leave 'finished' unset (open commit)
+		started = time.Now()
+	case started.IsZero() && !finished.IsZero():
+		// set 'started' to 'finished'
+		started = finished
+	case !started.IsZero() && finished.IsZero():
+		if now := time.Now(); now.Before(started) {
+			started = now // prevent finished < started (if user finishes commit soon)
+		}
+	case !started.IsZero() && !finished.IsZero():
+		if finished.Before(started) {
+			started = finished // prevent finished < started
```

Comment:
same as above, would we rather fail loudly here?

Reply:
As above, log a warning if this block is encountered